airbyte-cdk/python/airbyte_cdk/sources/streams/checkpoint/checkpoint_reader.py

```diff
@@ -111,19 +111,34 @@ def next(self) -> Optional[Mapping[str, Any]]:
         """
 
         try:
-            if self._current_slice is None:
-                self._current_slice = self._get_next_slice()
-                state_for_slice = self._cursor.select_state(self._current_slice)
-                if state_for_slice == {"__ab_full_refresh_sync_complete": True}:
-                    return None
-                else:
-                    return self._current_slice
             if self._read_state_from_cursor:
-                state_for_slice = self._cursor.select_state(self._current_slice)
-                if state_for_slice == {"__ab_full_refresh_sync_complete": True}:
-                    self._current_slice = self._get_next_slice()
+                if self._current_slice is None:
```

Comment:
nit: 
```python
if not self._current_slice:
    ...
```

Reply:
The difference here is subtle. But the intent here was that `StreamSlice` technically dervies from a mapping which could theoretically be an empty `{}` mapping.

To use a real example, let's say we have a basic stream that doesn't have any stream slicing and just returns a basic empty StreamSlice == {}.

The first iteration will work, but on the second page, we will enter this loop and move to the next slice which will return StopIteration. What we actually should do is move to the else and continue paginating.

So in this case `StreamSlice is None` is not equivalent to `not StreamSlice`